test/PowerShellEditorServices.Test/Session/ScriptFileTests.cs

```diff
@@ -580,5 +581,31 @@ public void PropertiesInitializedCorrectlyForUntitled()
                 Assert.Equal(3, scriptFile.FileLines.Count);
             }
         }
+
+        [Fact]
+        public void DocumentUriRetunsCorrectStringForAbsolutePath()
+        {
+            string path;
+            ScriptFile scriptFile;
+            var emptyStringReader = new StringReader("");
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                path = @"C:\Users\AmosBurton\projects\Rocinate\ProtoMolecule.ps1";
+                scriptFile = new ScriptFile(path, path, emptyStringReader, PowerShellVersion);
+                Assert.Equal("file:///c%3A/Users/AmosBurton/projects/Rocinate/ProtoMolecule.ps1", scriptFile.DocumentUri);
+            }
+            else
+            {
+                // Test the following only on Linux and macOS.
+                path = "/home/AmosBurton/projects/Rocinate/ProtoMolecule.ps1";
```

Comment:
what an obscure reference :P

Reply:
I'm all about obscure references.  :-)  Probably because I'm on book 7 now - been living with it for a while.  After we saw Lego Batman, all my Slack startup messages were Batman quotes.